Monika After Story/game/zz_windowreacts.rpy

```diff
@@ -0,0 +1,348 @@
+
+    #List of notif quips (used for topic alerts)
+    notif_quips = [
+        "[player], I want to talk to you about something.",
+        "Are you there, [player]?",
+        "Can you come here for a second?",
+        "[player], do you have a second?",
+        "I have something to tell you, [player]!",
+    ]
+
+    #List of hwnd IDs to destroy
+    destroy_list = list()
+
+    #START: Utility methods
+    def mas_getActiveWindow(friendly=False):
+        """
+        Gets the active window name
+
+        IN:
+            friendly: whether or not the active window name is returned in a state usable by the user
+        """
+        if (
+                renpy.windows
+                and mas_windowreacts.can_show_notifs
+                and persistent._mas_windowreacts_windowreacts_enabled
+            ):
+            from win32gui import GetWindowText, GetForegroundWindow
+
+            if not friendly:
+                return GetWindowText(GetForegroundWindow()).lower().replace(" ","")
+            else:
+                return GetWindowText(GetForegroundWindow()).lower()
+        else:
+            #TODO: Mac vers (if possible)
+            #NOTE: We return "" so this doesn't rule out notifications
+            return ""
+
+    def mas_isFocused():
+        """
+        Checks if MAS is the focused window
+        """
+        #TODO: Mac vers (if possible)
+        return store.mas_windowreacts.can_show_notifs and mas_getActiveWindow(True) == config.name.lower()
+
+    def mas_isInActiveWindow(keywords):
+        """
+        Checks if ALL keywords are in the active window name
+
+        IN:
+            List of keywords
+        """
+        active_window = mas_getActiveWindow()
+        return store.mas_windowreacts.can_show_notifs and len([s for s in keywords if s.lower() not in active_window]) == 0
+
+    def mas_clearNotifs():
+        """
+        Clears all tray icons (also action center on win10)
+        """
+        if renpy.windows and store.mas_windowreacts.can_show_notifs:
+            for index in range(len(destroy_list)-1,-1,-1):
+                win32gui.DestroyWindow(destroy_list[index])
+                destroy_list.pop(index)
+
+    def mas_checkForWindowReacts():
+        """
+        Runs through events in the windowreact_db to see if we have a reaction, and if so, queue it
+        """
+        #Do not check anything if we're not supposed to
+        if not persistent._mas_windowreacts_windowreacts_enabled or not store.mas_windowreacts.can_show_notifs:
+            return
+
+        for ev_label, ev in mas_windowreacts.windowreact_db.iteritems():
+            if (
+                    (mas_isInActiveWindow(ev.category) and ev.unlocked and ev.checkAffection(mas_curr_affection))
+                    and ((not store.mas_globals.in_idle_mode) or (store.mas_globals.in_idle_mode and ev.show_in_idle))
+                    and (not "winreact" in ev.rules or mas_notifsEnabledForGroup(ev.rules.get("winreact")))
```

Comment:
```suggestion
                    and ("winreact" not in ev.rules or mas_notifsEnabledForGroup(ev.rules.get("winreact")))
```

Reply:
looking at this, winreact, is actually a bad name for the key. should be `notif-group` as that is what it actually means.